fastapi/models/fastapi_endpoint.py

```diff
@@ -0,0 +1,268 @@
+            for rule in rec._make_routing_rule():
+                rec._endpoint_registry.add_or_update_rule(rule, init=init)
+
+    def _make_routing_rule(self):
+        """Generator of rule for every route into the routing info"""
+        self.ensure_one()
+        routing = self._get_routing_info()
+        for route in routing["routes"]:
+            key = self._endpoint_registry_route_unique_key(route)
+            endpoint_hash = hash(route)
+
+            def endpoint(self):
+                """Dummy method only used to register a route with type='fastapi'"""
+
+            endpoint.routing = routing
+            rule = self._endpoint_registry.make_rule(
+                key, route, endpoint, routing, endpoint_hash
+            )
+            yield rule
+
+    def _get_routing_info(self):
+        self.ensure_one()
+        return {
+            "type": "fastapi",
+            "auth": "public",
+            "methods": ["GET", "POST", "PUT", "DELETE", "OPTIONS"],
+            "routes": [
+                f"{self.root_path}/",
+                f"{self.root_path}/<path:application_path>",
+            ],
+            # csrf ?????
+        }
+
+    def _endpoint_registry_route_unique_key(self, route: str):
+        path = route.replace(self.root_path, "")
+        return f"{self._name}:{self.id}:{path}"
+
+    def _unregister_endpoints(self):
+        for rec in self:
+            for route in rec._get_routing_info()["routes"]:
+                key = rec._endpoint_registry_route_unique_key(route)
+                rec._endpoint_registry.drop_rule(key)
+
+    def _reset_app(self):
+        self.get_app.clear_cache(self)
+
+    @api.model
+    @tools.ormcache("root_path")
+    # TODO cache on thread local by db to enable to get 1 middelware by
+    # thread when odoo runs in multi threads mode and to allows invalidate
+    # specific entries in place og the overall cache as we have to do into
+    # the _rest_app method
+    def get_app(self, root_path):
+        record = self.search([("root_path", "=", root_path)])
+        if not record:
+            return None
+        return ASGIMiddleware(record._get_app())
+
+    @api.model
+    @tools.ormcache("root_path")
+    def get_uid(self, root_path):
+        record = self.search([("root_path", "=", root_path)])
+        if not record:
+            return None
+        return record.user_id.id
+
+    def _get_app(self) -> FastAPI:
+        app = FastAPI(**self._prepare_fastapi_endpoint_params())
+        for router in self._get_fastapi_routers():
+            app.include_router(prefix=self.root_path, router=router)
+        app.dependency_overrides[depends.fastapi_endpoint_id] = partial(
+            lambda a: a, self.id
+        )
+        for exception, handler in self._get_app_exception_handlers().items():
+            app.add_exception_handler(exception, handler)
+        return app
+
+    def _get_app_exception_handlers(
+        self,
+    ) -> Dict[
+        int | Type[Exception],
```

Comment:
```suggestion
        Union[int, Type[Exception]],
```

With this notation fastapi support python 3.7+ as you can see [documentation](https://fastapi.tiangolo.com/python-types/#union) section union. Otherwise we would be locked to python 3.10

Reply:
Thank you @flachica 
Happy to see someone making a try with this draft PR. I plan to work on it in the coming days to publish a first release.

Reply:
I am thoroughly studying this module and related technology. I intend to use it extensively in the very short term. I understand that it is in an initial state and I will do my best to contribute according to my possibilities and assuming that changes will come and I will have to adapt. Thank you very much for all the effort, especially in the documentation.

Reply:
I've to issues to solve to be complete. 
* The management of the languages to return the translated info into the requested language if available. This will be based on the accept-language http header
* The use of extendable-fastapi is not yet completely functional. For some processes, the fastapi lib use internal methods from the pydantic lib without going through the model class directly. This means that the extension mechanisms implemented by the extendable-fastapi library at the class definition level are bypassed in these cases.   I have identified at this stage 2 places that need to be adapted so that the schema for an extended model  is correctly generated in the documentation and the serialization is based on the extended class definition and not the base class. 

We'll probably find others issues but I'm confident that nothing should call into question the approach, which seems to me to be validated from a technical point of view. I'll start to use it for my own projects in the coming weeks.

Reply:
@flachica The integration of extendable-pydantic and fastapi is now fixed into the last version of extendable-pydantic (>=0.0.2) I've to adapt the documentation to specify that an extra odoo addon is required to 'extendable_fastapi'. I'll add the management of the languages tomorrow.